sycl/doc/extensions/proposed/sycl_ext_oneapi_queue_status_query.asciidoc

```diff
@@ -0,0 +1,125 @@
+
+
+== Notice
+
+[%hardbreaks]
+Copyright (C) 2022-2022 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 4 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+== Status
+
+This is a proposed extension specification, intended to gather community
+feedback.  Interfaces defined in this specification may not be implemented yet
+or may be in a preliminary state.  The specification itself may also change in
+incompatible ways before it is finalized.  *Shipping software products should
+not rely on APIs defined in this specification.*
+
+== Overview
+
+This extension adds `sycl::queue::size()`, `sycl::queue::empty()`, `sycl::queue::get_wait_list()` member functions.
+
+- `sycl::queue::size()` and `sycl::queue::empty` allow users to query the number of enqueued commands to a queue.
+- `sycl::queue::get_wait_list()` is the queue equivalent of `event::get_wait_list()` 
+
+== Specification
+
+=== Feature test macro
+
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_ONEAPI_QUEUE_STATUS_QUERY` to one of the values defined in the table
+below.  Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|Initial version of this extension.
+|===
+
+
+=== Signatures and Semantics 
+
+This extension adds the following new member functions to the SYCL `queue` class.
+```
+namespace sycl {
+namespace queue {
+ bool empty() const;
+ size_t size() const;
+ std::vector<event> get_wait_list() const;
+}}
+```
+
+The semantics of the new functions are: 
+
+```
+empty()
+   Returns `true` if `size() == 0`.
+```
+
+```
+size_t size() const
+   Returns the number of enqueued `commands` in the queue 
+   that have not been completed.
+   Since the implementation completes commands from the queue 
+   asynchronously, the returned value is a snapshot in time,
+   and the actual number of uncompleted commands may be different 
+   by the time the function returns.
+   Note that the behavior of `size()` is associated with the SYCL `queue`
+   and not with the underlying native-backend-object.
+```
+
+(based on similar HipSYCL extension https://github.com/illuhad/hipSYCL/blob/develop/doc/extensions.md#hipsycl_ext_queue_wait_list) 
```

Comment:
This is not a normative part of the specification, so it doesn't belong in the "Specification" section.  If you want to reference the HipSYCL extension, that could go to the "Overview" section or to some new section at the bottom (that is a peer of "Specification").